packages/squiggle-lang/src/rescript/Reducer/Reducer_Extension/Reducer_Extension_CodeTreeValue.res

```diff
@@ -0,0 +1,59 @@
+/*
+  Irreducable values. Reducer does not know about those. Only used for external calls
+  This is a configuration to to make external calls of those types
+*/
+module AE = Reducer_Extra_Array
+module Rerr = Reducer_Error
+
+type rec codeTreeValue =
```

Comment:
Extention is the part that needs to be edited to couple it with your distributions.
You are exdending the reducer.
It is not in CodeTree because, for CodeTree, the extensions are opaque. It does not want to know what is there. This is a guarantee that reducer code is not depending on the libraries of the language

Reply:
Maybe there's some other word we could call this besides "Extension"? "Extension" doesn't seem very descriptive at all, and this is a very fundamental thing. 

Related, I don't like the name ``ReducerLibrary``, as we already know it's in the reducer, and "Library" isn't very descriptive to me. 

It seems to me that both the CodeTreeValue and the ReducerLibrary both deal with two sides of ``CodeTreeValue``; I could see calling this directory just ``CodeTreeValue``, then having the "ReducerLibrary" the ``Dispatch``. 

Happy to discuss other names here.

Reply:
If the reasons are to keep the files I need to expose localized to one directory, I think that's a worthy design goal and I support the decision.

Reply:
@quinn-dougherty I'm not sure what you're referring to (which decision is "the decision"? )

Reply:
I was referring to this decision: 
> the extensions are opaque. It does not want to know what is there.

Reply:
RESOLVED